server/TelegramServerConfig.cpp

```diff
@@ -0,0 +1,107 @@
+        m_fileName = QStringLiteral("config.json");
+    } else {
+        m_fileName = fileName;
+    }
+
+    // default config
+    m_serverConfiguration.dcOptions = {
+        Telegram::DcOption(QStringLiteral("127.0.0.1"), 11441, 1),
+        Telegram::DcOption(QStringLiteral("127.0.0.2"), 11442, 2),
+        Telegram::DcOption(QStringLiteral("127.0.0.3"), 11443, 3),
+    };
+    m_privateKeyFile = QStringLiteral("private_key.pem");
+}
+
+bool Config::load()
+{
+    QByteArray bytes;
+    QFile f(m_fileName);
+    if (!f.open(QIODevice::ReadOnly)) {
+        qCWarning(loggingCategoryConfig) << "Failed to open config file:" << m_fileName;
+        return false;
+    }
+    bytes = f.readAll();
+    f.close();
+
+    QJsonParseError jErr;
+    const QJsonDocument doc = QJsonDocument::fromJson(bytes, &jErr);
+    if (jErr.error != QJsonParseError::NoError) {
+        qCWarning(loggingCategoryConfig) << "Failed to parse config file: " << jErr.errorString();
+        qCWarning(loggingCategoryConfig) << "Using default config.";
+        return false;
+    }
+
+    const QJsonObject obj = doc.object();
+    if (obj.isEmpty()) {
+        qCWarning(loggingCategoryConfig) << "Failed to parse config file: " << jErr.errorString();
+        qCWarning(loggingCategoryConfig) << "Using default config.";
+        return false;
+    }
+
+    // read private key setting
+    m_privateKeyFile = obj.value(QLatin1String("privateKeyFile")).toString();
+
+    // read server addr/port config
+    m_serverConfiguration.dcOptions.clear();
+    const QJsonArray &jserverConfig = obj.value(QLatin1String("serverConfiguration")).toArray();
+    for (const QJsonValue &jval: jserverConfig) {
+        const QJsonObject &jobj = jval.toObject();
+        DcOption dcOpt;
+        dcOpt.address = jobj.value(QLatin1String("address")).toString();
+        dcOpt.port = static_cast<quint16>(jobj.value(QLatin1String("port")).toInt());
+        m_serverConfiguration.dcOptions.append(dcOpt);
+    }
+
+    qCInfo(loggingCategoryConfig) << "Loaded config from " << m_fileName;
+    return true;
+}
+
+bool Config::save() const
+{
+    QJsonObject obj;
+    obj.insert(QStringLiteral("privateKeyFile"), m_privateKeyFile);
+
+    QJsonArray dcArr;
+    for (const DcOption &opt: m_serverConfiguration.dcOptions) {
+        QJsonObject jdcOpt;
+        jdcOpt.insert(QStringLiteral("address"), opt.address);
+        jdcOpt.insert(QStringLiteral("port"), opt.port);
+        dcArr.append(QJsonValue(jdcOpt));
+    }
+    obj.insert(QStringLiteral("serverConfiguration"), dcArr);
+
+    const QByteArray bytes = QJsonDocument(obj).toJson(QJsonDocument::Indented);
+
+    QFile f(m_fileName);
+    if (!f.open(QIODevice::WriteOnly)) {
+        qCWarning(loggingCategoryConfig) << "Failed to open config file:" << m_fileName;
+        return false;
+    }
+    qint64 written = f.write(bytes);
+    f.close();
+
+    return written == bytes.size();
+}
+
+} // Server
```

Comment:
Please update according to https://github.com/Kaffeine/telegram-qt/commit/92e033fea54ed9ccd37186987554c02418594e6c

Reply:
That commit only changes header files, none of `.cpp` files have it - [TelegramServerUser.cpp](https://github.com/Kaffeine/telegram-qt/blob/next/server/TelegramServerUser.cpp#L271)

Reply:
I fixed headers but source files should be fixed as well.
>none of `.cpp` files have it

In server only the [most recently added source](https://github.com/Kaffeine/telegram-qt/blob/3f340d6987d09155d4dcfb5e2844414fa1ca6cfa/server/DefaultAuthorizationProvider.cpp#L77) have it. However
```
TelegramQt$ grep "Telegram namespace" -R .|grep .cpp|wc -l
44
```

Reply:
:ok_hand: